src/dataset-builder/CohortEditor.ts

```diff
@@ -150,28 +152,43 @@ export const CriteriaView = ({ criteria, deleteCriteria, updateCriteria }: Crite
     ]
   );
 
-let criteriaCount = 1;
-const createDefaultListCriteria = (listOption: ProgramDataListOption): ProgramDataListCriteria => {
-  return {
-    kind: 'list',
-    listOption,
-    name: listOption.name,
-    id: Number(_.uniqueId('')),
-    count: 100,
-    values: [listOption.values[0]],
-  };
+const criteriaCount = atom(1);
+export const getNextCriteriaIndex = () => {
+  const localCriteriaCount = criteriaCount.get() + 1;
+  criteriaCount.set(localCriteriaCount);
+  return localCriteriaCount;
 };
```

Comment:
Ideally, we would track the criteriaIndex in the react state in a location that's shared between both the CohortEditor and DomainCriteriaSelector, which appears to be the DatasetBuilderView. But, I don't see a straightforward path to do this.

Reply:
That's a good point; right now this is all still a single page app so instead of using an atom store I could move this to state that's shared by both components. Moving into `DatasetBuilder` could work, let me try doing that.